### ISSUE TITLE
Bidirectional Forwarding Detection (BFD) support

### ISSUE THREAD

**Issue by jose-pavon-alvarez-db** [terraform-google-modules/terraform-google-cloud-router#45]
### TL;DR

BFD is supported on google_compute_router_peer since provider version 4.1.0.
Adding this configuration to both interconnect_attachment and interface submodules would allow to use this feature on interconnect cloud routers. Thanks.

### Terraform Resources

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_peer#example-usage---router-peer-bfd

### Detailed design

On the interconnect_attachment module:

```terraform
module "interconnect_attachment" {
...
peer = {
  name = "peer-name"
  peer_asn = "01234567"
  bfd = {
    min_receive_interval        = 1000
    min_transmit_interval       = 1000
    multiplier                  = 5
    session_initialization_mode = "ACTIVE"
  }
}
...
}
```

On the interface module, add a dynamic for bfd with the data provided by the peer variable (var.peer.bfd) if set. Here https://github.com/terraform-google-modules/terraform-google-cloud-router/blob/874ccd2001057722bbbb3f20ce42f9dadad4f074/modules/interface/main.tf#L39

### Additional information

Please update versions.tf on the interconnect_attachment module for google provider to be >= 4.1.0

**Comment by ivayloivanov388** [comment]
up
